templates/glossary.md

```diff
@@ -0,0 +1,253 @@
+
+### concrete structure
+
+### continuous integration
+
+
+
+### core Lean
+
+The part of Lean written in Lean itself. mathlib was originally an annex of Lean. Since then, it outgrew it. But some basic lemmas and definitions are still held in *core Lean*, [the Lean repository](https://github.com/leanprover-community/lean), which renders it much less flexible.
+
+### declaration
+
+A *declaration* is anything that starts with `def`, `lemma`, `theorem`. This corresponds to creating an (or several) object in the background environment.
+
+### defeq
+
+### delaborator
+
+### dependent type
+
+### dependent type theory
+
+### diamond
+
+There are often many ways to turn a given structure into another one. A *diamond* is a collection of such ways. Diamonds are abundant because of [hierarchies](#hierarchy). [Typeclass inference](#typeclass-inference) will unpredictably take one of the paths for a given diamond, so we want this path to not matter. This amounts to making the [Type-valued](#Prop-vs-Type) fields of the different inferable structures [defeq](#defeq). When this is the case, we have a *defeq diamond*.
+
+### elaborator
+
+### eliminator
+
+### equation compiler
+
+### `equiv`
+
+As distinct from mathematical equality,
+[`equiv`](mathlib_docs/data/equiv/basic.html) allows for defining an
+equivalence or congruence of types. One important thing to note
+is that an `equiv` [holds data instead of being merely a proof](#bundled-vs-unbundled).
+
+### eta reduction
+
+### `ext` lemma
+
+### heavy `refl`
+
+Some `refl` invokations take an obnoxiously long time. There can be many reasons for this. See [this Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/refl.20taking.2020.20seconds).
+
+### hierarchy
+
+A *hierarchy* is a collection of [typeclasses](#typeclass) which are more and more constraining. In mathlib, we have the *algebraic hierarchy* (`semiring`, `ring`, `field`, ...), the *order hierarchy* (`preorder`, `partial_order`, `linear_order`, ...), the *topological hierarchy* (`t1_space`, `t2_space`, `normal_space`, ...), the *categorical hierarchy* (`preadditive`, `abelian`, `monoidal`, ...) but also the *scalar hierarchy* (`mul_action`, `distrib_mul_action`, `module`, ...), the *norm hierarchy*, and intersection of previous ones like the *order-algebraic hierarchy*, the *topologico-algebraic hierarchy*, ...
+
+### `Icc`, `Ico`, `Ioc`, `Ioo`, `Ici`, `Ioi`, `Iic`, `Iio`
+
+There are `9` types of intervals, depending on whether the interval is **c**losed, **o**pen or runs to **i**nfinity at each end. Names have been made compact by calling each interval `I` + how it ends on the left + how it ends on the right.
```

Comment:
I think `Iii` is not really a thing right? So effectively there are only 8.

Reply:
Done.